pkg/scheduler/core/generic_scheduler.go

```diff
@@ -916,9 +916,11 @@ func filterPodsWithPDBViolation(pods []*v1.Pod, pdbs []*policy.PodDisruptionBudg
 				// We have found a matching PDB.
 				if pdbsAllowed[i] <= 0 {
 					pdbForPodIsViolated = true
-					break
 				} else {
-					pdbsAllowed[i]--
+					// If the pod exists in DisruptedPods, add the pod into nonViolatingPods directly.
```

Comment:
```suggestion
					// Only decrement the matched pdb when it's not in its <DisruptedPods>;
					// otherwise we may over-decrement the budget number.
```